ecommerce/enterprise/utils.py

```diff
@@ -776,3 +777,75 @@ def find_active_enterprise_customer_user(enterprise_customer_users):
         )
 
     return active_ecus[0]
+
+def get_enterprise_catalog_id_from_coupon(coupon_code):
+    """
+    Get the catalog ID associated with a given enterprise coupon code.
+
+    Args:
+        coupon_code (str): Code for the enterprise coupon.
+
+    Returns:
+        (int): Catalog ID associated with the given coupon code.
+        NoneType: Return None if the voucher does not exist or no catalog is found.
+    """
+    try:
+        voucher = Voucher.objects.get(code=coupon_code)
+    except Voucher.DoesNotExist:
+        log.warning(f"Voucher with code '{coupon_code}' does not exist")
+        return None
+
+    return _extract_catalog_id_from_voucher(voucher)
+
+
+def _extract_catalog_id_from_voucher(voucher):
+    """
+    Extract the catalog ID from a voucher's associated offers.
+
+    Args:
+        voucher (Voucher): The voucher object to extract the catalog ID from.
+
+    Returns:
+        (int): Catalog ID extracted from the voucher's offer condition range.
+        NoneType: Return None if no catalog is found or an error occurs during extraction.
+    """
+    try:
+        catalog_id = (
+            voucher.offers
+            .filter(condition__range__catalog_id__isnull=False)
+            .values_list('condition__range__catalog_id', flat=True)
+            .first()
+        )
+        if not catalog_id:
+            log.warning(f"No catalog found for voucher '{voucher.code}'")
+            return None
+        return catalog_id
+
+    except Exception as exc:
+        log.exception(
+            f"Error extracting catalog ID from voucher '{voucher.code}': {str(exc)}"
+        )
+        return None
+    
+def get_build_mfe_base_url(catalog_id):
+    """
+    Build the base URL for the ecommerce microfrontend receipt page with catalog parameter.
+
+    Args:
+        catalog_id (int): The catalog ID to include in the microfrontend URL.
+
+    Returns:
+        (str): The complete URL for the ecommerce microfrontend receipt page with the catalog parameter appended.
+        NoneType: Return None if MFE_REDIRECT_URL is not set in settings.
+    """
+    mfe_base_url = getattr(
+            settings,
+            'MFE_REDIRECT_URL',
+            None
+        )
```

Comment:
Since all MFEs use the LMS url as base URL I think that we can use this [model](https://github.com/Pearson-Advance/ecommerce/blob/607a43047b46e45fdc4550548d4447faa185f933/ecommerce/core/models.py#L39) which is the site configuration for each ecommerce site, obtain the lms_url_root and use it as the base URL to build the MFE URL, then the MFE_REDIRECT_URL setting would only be the path to the MFE, for example: '/coupons-mfe'. The site configuration object is present on every request, here is an example of usage in ecommerce: https://github.com/Pearson-Advance/ecommerce/blob/607a43047b46e45fdc4550548d4447faa185f933/ecommerce/extensions/basket/views.py#L546 https://github.com/Pearson-Advance/ecommerce/blob/607a43047b46e45fdc4550548d4447faa185f933/ecommerce/extensions/basket/views.py#L575